include/zephyr/bluetooth/audio/audio.h

```diff
@@ -199,26 +199,48 @@ struct bt_audio_codec_data {
+ *  @brief Helper to declare @ref bt_audio_codec_cfg
  *
  *  @param _id Codec ID
  *  @param _cid Company ID
  *  @param _vid Vendor ID
  *  @param _data Codec Specific Data in LVT format
  *  @param _meta Codec Specific Metadata in LVT format
  */
-#define BT_AUDIO_CODEC(_id, _cid, _vid, _data, _meta) \
-	{ \
-		/* Use HCI data path as default, can be overwritten by application */ \
-		.path_id = BT_ISO_DATA_PATH_HCI, \
-		.id = _id, \
-		.cid = _cid, \
-		.vid = _vid, \
-		.data_count = ARRAY_SIZE(((struct bt_audio_codec_data[])_data)), \
-		.data = _data, \
-		.meta_count = ARRAY_SIZE(((struct bt_audio_codec_data[])_meta)), \
-		.meta = _meta, \
-	}
+#define BT_AUDIO_CODEC_CFG(_id, _cid, _vid, _data, _meta)                                          \
+	((struct bt_audio_codec_cfg){                                                              \
+		/* Use HCI data path as default, can be overwritten by application */              \
+		.path_id = BT_ISO_DATA_PATH_HCI,                                                   \
+		.id = _id,                                                                         \
+		.cid = _cid,                                                                       \
+		.vid = _vid,                                                                       \
+		.data_count = ARRAY_SIZE(((struct bt_audio_codec_data[])_data)),                   \
+		.data = _data,                                                                     \
+		.meta_count = ARRAY_SIZE(((struct bt_audio_codec_data[])_meta)),                   \
+		.meta = _meta,                                                                     \
+	})
+
+/**
+ *  @brief Helper to declare @ref bt_audio_codec_cap structure
+ *
+ *  @param _id Codec ID
+ *  @param _cid Company ID
+ *  @param _vid Vendor ID
+ *  @param _data Codec Specific Data in LVT format
+ *  @param _meta Codec Specific Metadata in LVT format
+ */
+#define BT_AUDIO_CODEC_CAP(_id, _cid, _vid, _data, _meta)                                          \
+	((struct bt_audio_codec_cap){                                                              \
+		/* Use HCI data path as default, can be overwritten by application */              \
+		.path_id = BT_ISO_DATA_PATH_HCI,                                                   \
+		.id = (_id),                                                                       \
+		.cid = (_cid),                                                                     \
+		.vid = (_vid),                                                                     \
+		.data_len = sizeof((uint8_t[])_data),                                              \
+		.data = _data,                                                                     \
+		.meta_len = sizeof((uint8_t[])_meta),                                              \
+		.meta = _meta,                                                                     \
+	})
```

Comment:
Splitting this does not look right to me as Codec configuration might be non-LTV as well.
Thus if both capabilities and codec can be non-LTV the `.data` and `.meta` should be arrays for both cases meaning the representation of capabilities and codec should be identical.

Reply:
Essentially both have the same underlying data structure. 

The reason for splitting it, is to make it more clear in the API when to use which. 

Currently when we have a single structure, we have several places where codec capabilities have been passed to e.g. `bt_audio_codec_cfg_get_freq` which would use the wrong assigned numbers, or where a codec configuration has been registered as a capability. 

The main point of this is not to have two different underlying structures, but to make it clear to everyone when it is a capability structure (`bt_audio_codec_cap`) and when it is a configuration structure (`bt_audio_codec_cfg`). 

There is a PR that comes after this that makes the data and meta flat arrays as well: https://github.com/zephyrproject-rtos/zephyr/pull/58331